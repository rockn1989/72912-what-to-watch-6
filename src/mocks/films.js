export default [
  {
    id: 1,
    name: `The Grand Budapest Hotel`,
    posterImage: `img/the-grand-budapest-hotel-poster.jpg`,
    previewImage: `img/the-grand-budapest-hotel.jpg`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    previewVideolink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    scoresCount: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 99,
    genre: `Comedy`,
    realeased: 2014,
    isFavorite: false
  },
  {
    id: 2,
    name: `Aviator`,
    posterImage: `img/aviator.jpg`,
    previewImage: `img/aviator.jpg`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    previewVideolink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 2.9,
    scoresCount: 20,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 99,
    genre: `Advanture`,
    realeased: 2014,
    isFavorite: false
  },
  {
    id: 3,
    name: `Revenant`,
    posterImage: `img/revenant.jpg`,
    previewImage: `img/revenant.jpg`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    previewVideolink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 6.9,
    scoresCount: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 99,
    genre: `Comedy`,
    realeased: 2014,
    isFavorite: false
  },
  {
    id: 4,
    name: `Johnny English`,
    posterImage: `img/johnny-english.jpg`,
    previewImage: `img/johnny-english.jpg`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    previewVideolink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    scoresCount: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 99,
    genre: `Comedy`,
    realeased: 2014,
    isFavorite: false
  },
  {
    id: 5,
    name: `What we do is the shadow`,
    posterImage: `img/what-we-do-in-the-shadows.jpg`,
    previewImage: `img/what-we-do-in-the-shadows.jpg`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    previewVideolink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 3.9,
    scoresCount: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 99,
    genre: `Comedy`,
    realeased: 2014,
    isFavorite: false
  },
  {
    id: 6,
    name: `We need to talk about Kevin`,
    posterImage: `img/we-need-to-talk-about-kevin.jpg`,
    previewImage: `img/we-need-to-talk-about-kevins.jpg`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    previewVideolink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    scoresCount: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 99,
    genre: `Drama`,
    realeased: 2009,
    isFavorite: false
  },
  {
    id: 7,
    name: `Macbeth`,
    posterImage: `img/macbeth.jpg`,
    previewImage: `img/macbeth.jpg`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    previewVideolink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    scoresCount: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 99,
    genre: `Drama`,
    realeased: 2014,
    isFavorite: false
  },
  {
    id: 8,
    name: `Bohemain Rhaapsody`,
    posterImage: `img/bohemian-rhapsody.jpg`,
    previewImage: `img/bohemian-rhapsody.jpg`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    previewVideolink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 9.9,
    scoresCount: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 99,
    genre: `History`,
    realeased: 2014,
    isFavorite: false
  },
  {
    id: 9,
    name: `We need to talk about Kevin`,
    posterImage: `img/we-need-to-talk-about-kevin.jpg`,
    previewImage: `img/we-need-to-talk-about-kevins.jpg`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    previewVideolink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    scoresCount: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 99,
    genre: `Drama`,
    realeased: 2009,
    isFavorite: false
  },
  {
    id: 10,
    name: `Macbeth`,
    posterImage: `img/macbeth.jpg`,
    previewImage: `img/macbeth.jpg`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    previewVideolink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    scoresCount: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 99,
    genre: `Drama`,
    realeased: 2014,
    isFavorite: false
  },
  {
    id: 11,
    name: `Bohemain Rhaapsody`,
    posterImage: `img/bohemian-rhapsody.jpg`,
    previewImage: `img/bohemian-rhapsody.jpg`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    previewVideolink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 9.9,
    scoresCount: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 99,
    genre: `History`,
    realeased: 2014,
    isFavorite: false
  },
];
