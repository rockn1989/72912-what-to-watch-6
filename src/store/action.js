export const ActionType = {
  SET_ACTIVE_GENRE: `set_genre`,
  FILTER_FILMS: `filter_films`
};

export const ActionCreator = {
  setGenre: (genre) => ({
    type: ActionType.SET_ACTIVE_GENRE,
    payload: genre
  }),
  filterFilms: () => ({
    type: ActionType.FILTER_FILMS
  })
};
