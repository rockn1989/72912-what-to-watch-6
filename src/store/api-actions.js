import {
  loadFilmsAction,
  loadFilmAction,
  getFavoriteAction,
  setFavoriteAction,
  redirectAction,
  authorizationAction,
  setUserInfoAction,
  sendFormDataAction,
  showErrorAction
} from "./action";

export const loadFilmsList = () => (dispatch, _getState, api) => (
  api.get(`/films`)
    .then(({data}) => {
      dispatch(loadFilmsAction(data));
    })
);

export const loadFilm = (id) => (dispatch, _getState, api) => (
  api.get(`/films/${id}`)
    .then(({data}) => {
      dispatch(loadFilmAction(data));
    })
    .catch(() => {
      dispatch(redirectAction(`/page-not-found`));
    })
);

export const getFavoriteFilms = () => (dispatch, _getState, api) => (
  api.get(`/favorite`)
    .then(({data}) => {
      dispatch(getFavoriteAction(data));
    })
);

export const setFavoriteFilm = (id, status) => (dispatch, _getState, api) => (
  api.post(`/favorite/${id}/${status}`)
    .then(({data}) => {
      dispatch(setFavoriteAction(data));
    })
);

export const checkAuth = () => (dispatch, _getState, api) => (
  api.get(`/login`)
    .then(({data}) => {
      dispatch(authorizationAction(true));
      dispatch(setUserInfoAction(data));
    })
);

export const logout = () => (dispatch, _getState, api) => (
  api.get(`/logout`)
    .then(({status}) => {
      if (status === 200) {
        dispatch(authorizationAction(false));
      }
    })
);

export const sendLogin = ({email, password}) => (dispatch, _getState, api) => (
  api.post(`/login`, {
    email,
    password
  })
  .then(({data}) => {
    dispatch(setUserInfoAction(data));
    dispatch(authorizationAction(true));
    dispatch(redirectAction(`/`));
  })
  .catch(() => {
    dispatch(authorizationAction(false));
  })
);

export const sendComment = ({id, comment, rating}) => (dispatch, _getState, api) => {
  dispatch(sendFormDataAction(false));
  return api.post(`/comments/${id}`, {
    rating,
    comment
  }).then(() => {
    dispatch(sendFormDataAction(true));
    dispatch(redirectAction(`/films/${id}`));
  })
  .catch(() => {
    dispatch(showErrorAction(true));
  });
};
