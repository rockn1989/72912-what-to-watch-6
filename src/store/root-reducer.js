import {combineReducers} from 'redux';

import {filmsData} from './films-data/films-data';
import {filmData} from './film-data/film-data';
import {user} from './user/user';
import {formStatus} from './form-status/form-status';
import {errorStatus} from './error-status/error-status';

export const NameSpace = {
  FILMS_DATA: `FILMS_DATA`,
  FILM_DATA: `FILM_DATA`,
  USER: `USER`,
  FORM_STATUS: `FORM_STATUS`,
  ERROR_STATUS: `ERROR_STATUS`
};

export default combineReducers({
  [NameSpace.FILMS_DATA]: filmsData,
  [NameSpace.FILM_DATA]: filmData,
  [NameSpace.USER]: user,
  [NameSpace.FORM_STATUS]: formStatus,
  [NameSpace.ERROR_STATUS]: errorStatus,
});

// export default combineReducers({
//   filmsData,
//   genre,
//   user,
//   formStatus,
//   errorStatus,
// });
