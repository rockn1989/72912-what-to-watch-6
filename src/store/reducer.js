import {ActionType} from './action';
import {AUTH, NO_AUTH} from '../const';

const initialState = {
  genre: `All genres`,
  film: {},
  films: [],
  filtredFilmsList: [],
  filmsCounter: [],
  authorizationStatus: NO_AUTH,
  userInfo: {
    avatarUrl: `img/avatar.jpg`
  }
};

const reducer = (state = initialState, action) => {

  switch (action.type) {
    case ActionType.LOAD_FILMS:
      return {
        ...state,
        films: action.payload
      };

    case ActionType.LOAD_FILM:
      return {
        ...state,
        film: action.payload
      };

    case ActionType.SET_ACTIVE_GENRE:
      return {
        ...state,
        genre: action.payload
      };

    case ActionType.FILTER_FILMS:
      return {
        ...state,
        filtredFilmsList: state.genre === initialState.genre ? [...state.films] : state.films.filter((film) => film.genre === state.genre)
      };

    case ActionType.SHOW_MORE:
      const genreCounter = state.filmsCounter.filter(({name}) => name === state.genre);
      const idx = state.filmsCounter.indexOf(...genreCounter);

      const newEl = {
        name: state.genre,
        counter: genreCounter[0].counter + 1
      };

      const newFilmsCounter = [...state.filmsCounter.slice(0, idx), newEl, ...state.filmsCounter.slice(idx + 1)];
      return {
        ...state,
        filmsCounter: [...newFilmsCounter]
      };

    case ActionType.SET_FILMS_COUNTER:
      const counters = [];
      for (let name of action.payload) {
        counters.push({
          name,
          counter: 0
        });
      }

      return {
        ...state,
        filmsCounter: [...counters]
      };

    case ActionType.RESET_FILMS_COUNTER:
      const resetFilmsCounter = state.filmsCounter.map((counter) => {
        return {
          name: counter.name,
          counter: 0
        };
      });
      return {
        ...state,
        filmsCounter: [...resetFilmsCounter]
      };

    case ActionType.AUTHORIZED:
      return {
        ...state,
        authorizationStatus: action.payload ? AUTH : NO_AUTH
      };

    case ActionType.SET_USER_INFO:
      const {avatar_url: avatar, ...restData} = {...action.payload};
      return {
        ...state,
        userInfo: {...restData, avatarUrl: avatar}
      };
  }
  return state;
};

export {reducer};
