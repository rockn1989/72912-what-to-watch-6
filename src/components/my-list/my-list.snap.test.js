import React from 'react';
import {render} from '@testing-library/react';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import * as redux from 'react-redux';
import configureStore from 'redux-mock-store';
import {NameSpace} from '../../store/root-reducer';

import MyList from './my-list';

import films from '../../mocks/films';
const mockStore = configureStore({});

const store = mockStore({
  [NameSpace.USER]: {
    authorizationStatus: false,
    userInfo: {
      avatarUrl: `img/avatar.jpg`
    }
  },
  [NameSpace.FILMS_DATA]: {
    genre: `All genres`,
    filmsCounter: [],
    films,
    favorites: []
  },
  [NameSpace.FILM_DATA]: {
    film: films[0]
  },
  [NameSpace.FORM_STATUS]: {
    formStatus: true
  },
  [NameSpace.ERROR_STATUS]: {
    error: false
  }
});

it(`Should MyList render correctly`, () => {
  jest.spyOn(redux, `useSelector`);
  const history = createMemoryHistory();

  const {container} = render(
      <redux.Provider store={store}>
        <Router history={history}>
          <MyList onGetFavoriteFilms={jest.fn()} />
        </Router>
      </redux.Provider>
  );

  expect(container).toMatchSnapshot();
});
