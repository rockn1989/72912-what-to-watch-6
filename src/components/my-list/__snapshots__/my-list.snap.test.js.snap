// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MyList render correctly 1`] = `
<div>
  <div
    class="user-page"
  >
    <header
      class="page-header user-page__head"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <h1
        class="page-title user-page__title"
      >
        My list
      </h1>
      <div
        class="user-block"
      >
        <a
          class="user-block__link"
          href="/login"
        >
          Sign in
        </a>
      </div>
    </header>
    <section
      class="catalog"
    >
      <h2
        class="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <div
        class="catalog__movies-list"
      />
    </section>
    <footer
      class="page-footer"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link logo__link--light"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
